jobs/periodic/flag-old-conversations/index.ts

```diff
@@ -0,0 +1,94 @@
+import moment from 'moment';
+import { getMatch, getSubcourse } from '../../../graphql/util';
+import { getAllConversations, markConversationAsReadOnly, sendSystemMessage } from '../../../common/chat';
+import { getLogger } from '../../../common/logger/logger';
+import { Lecture } from '../../../graphql/generated';
+import { SystemMessage } from '../../../common/chat/types';
+import systemMessages from '../../../common/chat/localization';
+
+const logger = getLogger();
+
+enum ConversationType {
+    GROUP = 'group',
+    ONE_ON_ONE = 'one_on_one',
+}
+
+type conversationsToDeactivate = {
+    id: string;
+    conversationType: ConversationType;
+};
+async function isActiveMatch(id: number): Promise<boolean> {
+    const today = moment().endOf('day');
+    const match = await getMatch(id);
+    const dissolvedAtPlus30Days = moment(match.dissolvedAt).add(30, 'days');
+    return !dissolvedAtPlus30Days.isBefore(today);
+}
+
+async function isActiveSubcourse(id: number): Promise<boolean> {
+    const today = moment().endOf('day');
+    const subcourse = await getSubcourse(id, true);
+    const isSubcourseCancelled = subcourse.cancelled;
+
+    if (isSubcourseCancelled) {
+        return false;
+    } else {
+        const lastLecutre = subcourse.lecture.sort((a: Lecture, b: Lecture) => moment(a.start).milliseconds() - moment(b.start).milliseconds()).pop();
+        const lastLecturePlus30Days = moment(lastLecutre.start).add(30, 'days');
+        const is30DaysBeforeToday = lastLecturePlus30Days.isBefore(today);
+        return !is30DaysBeforeToday;
+    }
+}
+
+export default async function flagInactiveConversationsAsReadonly() {
+    const conversations = await getAllConversations();
```

Comment:
Can we query only active conversations, so that this doesn't steadily grow over time?